src/zkp/pisch.rs

```diff
@@ -200,18 +205,20 @@ impl PiSchProof {
     }
     pub(crate) fn from_message(message: &Message) -> Result<Self> {
         message.check_type(MessageType::Keygen(KeygenMessageType::R3Proof))?;
-        let keygen_decommit: PiSchProof = deserialize!(&message.unverified_bytes)?;
-        Ok(keygen_decommit)
+        let pischproofbytes: PiSchProof = deserialize!(&message.unverified_bytes)?;
+        assert!(pischproofbytes.challenge < k256_order());
+        //assert!(pischproofbytes.response < k256_order());
```

Comment:
todo: this will panic if it fails! Use a conditional and return an error (probably `ProtocolError`) instead.

also, question: why is the response check commented out?

Reply:
Aha. Looking at how we generate this, I think the idea you brought up in the team meeting is correct -- when we compute `response`, we use normal addition and multiplication instead of the modular +/* with `BigNumber`s. So it makes sense that it's too big. This is a bug; the paper explicitly says to calculate it `mod q`.

Do you want to change it here or in a separate issue? It should just be one line change  in `prove_from_precommit` to use `modadd` and `modmul` instead of `+` and `*`.

Reply:
changed in this PR itself.